sycl/plugins/native_cpu/pi_native_cpu.cpp

```diff
@@ -1,1411 +1,172 @@
-                                          size_t, size_t, size_t, pi_uint32,
-                                          const pi_ext_sync_point *,
-                                          pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextCommandBufferMemBufferCopyRect(
-    pi_ext_command_buffer, pi_mem, pi_mem, pi_buff_rect_offset,
-    pi_buff_rect_offset, pi_buff_rect_region, size_t, size_t, size_t, size_t,
-    pi_uint32, const pi_ext_sync_point *, pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextCommandBufferMemBufferRead(pi_ext_command_buffer, pi_mem, size_t,
-                                          size_t, void *, pi_uint32,
-                                          const pi_ext_sync_point *,
-                                          pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextCommandBufferMemBufferReadRect(
-    pi_ext_command_buffer, pi_mem, pi_buff_rect_offset, pi_buff_rect_offset,
-    pi_buff_rect_region, size_t, size_t, size_t, size_t, void *, pi_uint32,
-    const pi_ext_sync_point *, pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextCommandBufferMemBufferWrite(pi_ext_command_buffer, pi_mem,
-                                           size_t, size_t, const void *,
-                                           pi_uint32, const pi_ext_sync_point *,
-                                           pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextCommandBufferMemBufferWriteRect(
-    pi_ext_command_buffer, pi_mem, pi_buff_rect_offset, pi_buff_rect_offset,
-    pi_buff_rect_region, size_t, size_t, size_t, size_t, const void *,
-    pi_uint32, const pi_ext_sync_point *, pi_ext_sync_point *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextEnqueueCommandBuffer(pi_ext_command_buffer, pi_queue, pi_uint32,
-                                    const pi_event *, pi_event *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextEnablePeerAccess(pi_device, pi_device) { DIE_NO_IMPLEMENTATION; }
-
-pi_result piextDisablePeerAccess(pi_device, pi_device) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piextPeerAccessGetInfo(pi_device, pi_device, pi_peer_attr, size_t,
-                                 void *, size_t *) {
-  DIE_NO_IMPLEMENTATION;
-}
-
-pi_result piTearDown(void *) {
-  // Todo: is it fine as a no-op?
-  return PI_SUCCESS;
-}
 
 pi_result piPluginInit(pi_plugin *PluginInit) {
-#define _PI_API(api)                                                           \
-  (PluginInit->PiFunctionTable).api = (decltype(&::api))(&api);
-#include <sycl/detail/pi.def>
 
+#define _PI_CL(pi_api, native_cpu_api)                                         \
+  (PluginInit->PiFunctionTable).pi_api = (decltype(&::pi_api))(&native_cpu_api);
+
+  // Platform
+  _PI_CL(piPlatformsGet, pi2ur::piPlatformsGet)
+  _PI_CL(piPlatformGetInfo, pi2ur::piPlatformGetInfo)
+  _PI_CL(piPluginGetBackendOption, pi2ur::piPluginGetBackendOption)
```

Comment:
We recently discovered that this approach results in an ABI break of the PI plugin exported symbols, please see patches in https://github.com/intel/llvm/pull/10518 for the recommended way to avoid this ABI break.

Reply:
Thanks. We'll add a similar check for this in a subsequent PR since this part is currently WIP and no current tests are broken. Would that be ok?

Reply:
As long as that change is included in the 2024.0 release 👍

Reply:
We'll have another PR up this or next week 👍